crates/protogen/Cargo.toml

```diff
@@ -9,15 +9,15 @@ edition.workspace = true
 datafusion = { workspace = true }
 datafusion-proto = { workspace = true }
 thiserror.workspace = true
-tonic = { version = "0.9", features = ["transport", "tls", "tls-roots"] }
-prost = "0.11"
-prost-types = "0.11"
+tonic = { workspace = true }
+prost = { workspace = true }
+prost-types = { workspace = true }
 proptest = "1.2"
 proptest-derive = "0.3"
 uuid = { version = "1.4.1", features = ["v4", "fast-rng", "macro-diagnostics"] }
 tracing = "0.1"
 
 [build-dependencies]
-tonic-build = "0.9"
+tonic-build = "0.10"
 # Only needed to handle custom btree mapping; can be removed when we bump the tonic version which will have this too
-prost-build = "0.11"
+prost-build = "0.12"
```

Comment:
can these be workspace too?

Reply:
Possibly, I haven't tried doing workspace dev dependencies, but I imagine it would work.

(Also Vaibhav is taking over this pr)